airbyte-integrations/base-java/src/main/java/io/airbyte/integrations/base/IntegrationCliParser.java

```diff
@@ -0,0 +1,193 @@
+          .builder().longOpt(JavaBaseConstants.ARGS_CONFIG_KEY).desc(JavaBaseConstants.ARGS_CONFIG_DESC).hasArg(true).required(true).build());
+      case READ -> {
+        options.addOption(Option
+            .builder().longOpt(JavaBaseConstants.ARGS_CONFIG_KEY).desc(JavaBaseConstants.ARGS_CONFIG_DESC).hasArg(true).required(true).build());
+        options.addOption(Option
+            .builder().longOpt(JavaBaseConstants.ARGS_SCHEMA_KEY).desc(JavaBaseConstants.ARGS_SCHEMA_DESC).hasArg(true).build());
+        options.addOption(Option
+            .builder().longOpt(JavaBaseConstants.ARGS_STATE_KEY).desc(JavaBaseConstants.ARGS_PATH_DESC).hasArg(true).build());
+      }
+      case WRITE -> {
+        options.addOption(Option
+            .builder().longOpt(JavaBaseConstants.ARGS_CONFIG_KEY).desc(JavaBaseConstants.ARGS_CONFIG_DESC).hasArg(true).required(true).build());
+        options.addOption(Option
+            .builder().longOpt(JavaBaseConstants.ARGS_SCHEMA_KEY).desc(JavaBaseConstants.ARGS_SCHEMA_DESC).hasArg(true).build());
+      }
+      default -> throw new IllegalStateException("Unexpected value: " + command);
+    }
+
+    final CommandLine parsed = runParse(options, args, command);
+    Preconditions.checkNotNull(parsed);
+    final Map<String, String> argsMap = new HashMap<>();
+    for (final Option option : parsed.getOptions()) {
+      argsMap.put(option.getLongOpt(), option.getValue());
+    }
+    LOGGER.info("integration args: {}", argsMap);
+
+    switch (command) {
+      case SPEC -> {
+        return IntegrationConfig.spec();
+      }
+      case CHECK -> {
+        return IntegrationConfig.check(argsMap.get(JavaBaseConstants.ARGS_CONFIG_KEY));
+      }
+      case DISCOVER -> {
+        return IntegrationConfig.discover(argsMap.get(JavaBaseConstants.ARGS_CONFIG_KEY));
+      }
+      case READ -> {
+        return IntegrationConfig.read(
+            argsMap.get(JavaBaseConstants.ARGS_CONFIG_KEY),
+            argsMap.get(JavaBaseConstants.ARGS_SCHEMA_KEY),
+            argsMap.get(JavaBaseConstants.ARGS_STATE_KEY));
+      }
+      case WRITE -> {
+        return IntegrationConfig.write(
+            argsMap.get(JavaBaseConstants.ARGS_CONFIG_KEY),
+            argsMap.get(JavaBaseConstants.ARGS_SCHEMA_KEY));
+      }
+      default -> throw new IllegalStateException("Unexpected value: " + command);
+    }
+  }
+
+  private static CommandLine runParse(Options options, String[] args, Command command) {
+    final CommandLineParser parser = new DefaultParser();
+    final HelpFormatter helpFormatter = new HelpFormatter();
+
+    try {
+      return parser.parse(options, args);
+    } catch (ParseException e) {
+      helpFormatter.printHelp(command.toString().toLowerCase(), options);
+      throw new RuntimeException(e);
+    }
+  }
+
+  // https://stackoverflow.com/questions/33874902/apache-commons-cli-1-3-1-how-to-ignore-unknown-arguments
+  private static class RelaxedParser extends DefaultParser {
```

Comment:
Why do we need that?

Reply:
see this discussion: https://github.com/airbytehq/airbyte/pull/491#discussion_r500553058